interface/ceed-qfunction.c

```diff
@@ -328,6 +387,65 @@ int CeedQFunctionGetInnerContext(CeedQFunction qf, CeedQFunctionContext *ctx) {
   return CEED_ERROR_SUCCESS;
 }
 
+/**
+  @brief Get inner context data of a CeedQFunction
+
+  @param qf         CeedQFunction
+  @param mem_type   Memory type on which to access the data. If the backend
+                      uses a different memory type, this will perform a copy.
+  @param[out] data  Data on memory type mem_type
+
+  @return An error code: 0 - success, otherwise - failure
+
+  @ref Backend
+**/
+int CeedQFunctionGetInnerContextData(CeedQFunction qf, CeedMemType mem_type,
+                                     void *data) {
+  int ierr;
+  bool is_writable;
+  CeedQFunctionContext ctx;
+
+  ierr = CeedQFunctionGetInnerContext(qf, &ctx); CeedChk(ierr);
+  if (ctx) {
+    ierr = CeedQFunctionIsContextWritable(qf, &is_writable); CeedChk(ierr);
+    if (is_writable) {
+      ierr = CeedQFunctionContextGetData(ctx, mem_type, data); CeedChk(ierr);
+    } else {
+      ierr = CeedQFunctionContextGetDataRead(ctx, mem_type, data); CeedChk(ierr);
+    }
```

Comment:
It looks like the default will be `is_context_writable = false` so we'll act as though it's read-only even though the user may write to it. Should the default be writable and the user needs to mark it if not (for the performance benefit)? Or should we make it explicit when the context is first set (`CeedQFunctionSetContextWritable`)?